src/firefly/js/charts/ui/ColumnOrExpression.jsx

```diff
@@ -109,10 +110,9 @@ export function ColumnFld({cols, groupKey, fieldKey, initValue, label, tooltip='
             <ToolbarButton icon={MAGNIFYING_GLASS}
                            tip={`Select ${name} column`}
                            onClick={(e) => {
-                               if (!onSearchClicked || onSearchClicked()) {
-                                   showColSelectPopup(cols, onColSelected, `Choose ${name}`, 'OK',
-                                       val, false, colTblId,doQuoteNonAlphanumeric);
-                               }
+                               onSearchBtnClicked?.(e);
+                               showColSelectPopup(cols, onColSelected, `Choose ${name}`, 'OK',
+                                       val, false, colTblId, doQuoteNonAlphanumeric);
```

Comment:
before `onSearchClicked` has to exist to call `showColSelectPopup`. That seems like it was probably a bug.

Reply:
This definitely changed the logic.  Original logic is to skip `showColSelectPopup` when `onSearchBtnClicked` returns false.  I am guessing it's a way to do some sort of validation.  @kpuriIpac should look at this.

Reply:
I checked uses of `onSearchClicked` in our codebase before changing it. It was being used only in src/firefly/js/ui/UploadTableSelectorSingleCol.jsx L156-160, and it always returns true (see the deleted file below)

I did find it confusing though that how it was maintaining a toggle flag in the state to do so which didn't seem useful. So @kpuriIpac should confirm if I took out something he wanted to keep?

Reply:
Yes I can't seem to recall the exact reason for it being there, but you're right that it always returned true @jaladh-singhal, probably a bug anyway. This change looks good and everything still works as expected.

Reply:
@kpuriIpac ok - thanks for testing!